code/modules/hydroponics/hydro_tray.dm

```diff
@@ -643,6 +643,44 @@
 		anchored = !anchored
 		to_chat(user, "You [anchored ? "wrench" : "unwrench"] \the [src].")
 
+/obj/structure/machinery/portable_atmospherics/hydroponics/get_examine_text(mob/user)
+	. = ..()
+	var/hyrdro_info = show_hydro_info(user)
+	if(hyrdro_info)
+		. += hyrdro_info
+
+
+/obj/structure/machinery/portable_atmospherics/hydroponics/proc/show_hydro_info(mob/user as mob)
+	var/info = ""
+	if(seed && !dead)
+		info += "[src] has \blue [seed.display_name] \black planted.\n"
+		if(plant_health <= (seed.endurance / 2))
+			info += "The plant looks \red unhealthy.\n"
+	else
+		info += "[src] is empty.\n"
+	info += "Water: [round(waterlevel,0.1)]/100\n"
+	info += "Nutrient: [round(nutrilevel,0.1)]/10\n"
+	if(weedlevel >= 5)
+		info += "[src] is \red filled with weeds!\n"
+	if(pestlevel >= 5)
+		info += "[src] is \red filled with tiny worms!\n"
+
+	if(!istype(src,/obj/structure/machinery/portable_atmospherics/hydroponics/soil))
```

Comment:
i know you're just moving this code, but please make this not do istype(src), just make a proc override or something